test/integration/apiserver/anonymous/anonymous_test.go

```diff
@@ -0,0 +1,332 @@
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+	"k8s.io/component-base/featuregate"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+	kubeapiserverapptesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
+	"k8s.io/kubernetes/pkg/apis/rbac"
+	"k8s.io/kubernetes/test/integration/framework"
+)
+
+const (
+	defaultNamespace           = "default"
+	defaultRBACRoleName        = "developer-role"
+	defaultRBACRoleBindingName = "developer-role-binding"
+)
+
+var (
+	defaultRole = &rbacv1.Role{
+		TypeMeta:   metav1.TypeMeta{APIVersion: "rbac.authorization.k8s.io/v1", Kind: "Role"},
+		ObjectMeta: metav1.ObjectMeta{Name: defaultRBACRoleName},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs:         []string{"list"},
+				Resources:     []string{"pods"},
+				APIGroups:     []string{""},
+				ResourceNames: []string{},
+			},
+		},
+	}
+	defaultRoleBinding = &rbacv1.RoleBinding{
+		TypeMeta:   metav1.TypeMeta{APIVersion: "rbac.authorization.k8s.io/v1", Kind: "RoleBinding"},
+		ObjectMeta: metav1.ObjectMeta{Name: defaultRBACRoleBindingName},
+		Subjects: []rbacv1.Subject{
+			{
+				APIGroup: rbac.GroupName,
+				Kind:     rbacv1.UserKind,
+				Name:     "system:anonymous",
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbac.GroupName,
+			Kind:     "Role",
+			Name:     defaultRBACRoleName,
+		},
+	}
+)
+
+func TestStructuredAuthenticationConfig(t *testing.T) {
+	t.Log("Testing anonymous authenticator with authentication config")
+
+	testCases := []struct {
+		desc            string
+		featureGates    map[featuregate.Feature]bool
+		authConfig      string
+		additionalFlags []string
+		assertErrFn     func(t *testing.T, errorToCheck error)
+		assertFn        func(t *testing.T, server kubeapiserverapptesting.TestServer)
+	}{
+		{
+			desc: "StructuredAuthenticationConfiguration disabled",
+			featureGates: map[featuregate.Feature]bool{
+				features.StructuredAuthenticationConfiguration: false,
+			},
+			authConfig: `
+apiVersion: apiserver.config.k8s.io/v1beta1
+kind: AuthenticationConfiguration
+anonymous:
+  enabled: true
+  conditions:
+  - path: "/livez"
+`,
+			assertErrFn: func(t *testing.T, errorToCheck error) {
+				assert.EqualError(t, errorToCheck, "failed to validate ServerRunOptions: set --feature-gates=StructuredAuthenticationConfiguration=true to use authentication-config file")
+			},
+		},
+		{
+			desc: "AnonymousAuthConfigurableEndpoints disabled",
+			featureGates: map[featuregate.Feature]bool{
+				features.StructuredAuthenticationConfiguration: true,
+				features.AnonymousAuthConfigurableEndpoints:    false,
+			},
+			authConfig: `
+apiVersion: apiserver.config.k8s.io/v1beta1
+kind: AuthenticationConfiguration
+anonymous:
+  enabled: true
+  conditions:
+  - path: "/livez"
+`,
+			assertErrFn: func(t *testing.T, errorToCheck error) {
+				assert.EqualError(t, errorToCheck, "anonymous: Forbidden: anonymous is not supported when when AnonymousAuthConfigurableEnpoints feature gate is disabled")
```

Comment:
typo "when when".

the message is a bit confusing:
> anonymous is not supported ...

should that be:
> anonymous auth is not supported ...

Reply:
updated. Thanks!